packages/components/src/tabs/tab.tsx

```diff
@@ -2,24 +2,26 @@
  * WordPress dependencies
  */
 
-import { useContext, forwardRef } from '@wordpress/element';
+import { forwardRef } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import type { TabProps } from './types';
 import warning from '@wordpress/warning';
-import { TabsContext } from './context';
+import { useTabsContext } from './context';
 import { Tab as StyledTab } from './styles';
 import type { WordPressComponentProps } from '../context';
 
 export const Tab = forwardRef<
 	HTMLButtonElement,
 	WordPressComponentProps< TabProps, 'button', false >
 >( function Tab( { children, id, disabled, render, ...otherProps }, ref ) {
-	const context = useContext( TabsContext );
+	const context = useTabsContext();
 	if ( ! context ) {
-		warning( '`Tabs.TabList` must be wrapped in a `Tabs` component.' );
+		warning(
+			'`Tabs.Tab` needs to receive context from a `Tabs` component or a `Tabs.Context.Provider`.'
+		);
```

Comment:
This warning could be quite confusing, because `Tabs.Tab` should be always be wrapped in a `Tabs` component (wrapping it in `Tabs.Context.Provider` is not an alternative to wrapping it in `Tabs`, but rather a band-aid for when the context is not forwarded correctly, usually because of slot/fills).

Probably keeping the original warning message makes more sense to me (this applies to all warning message changes in this PR)